packages/next/src/seo/index.ts

```diff
@@ -0,0 +1,163 @@
+import {
+	generateRootQuery,
+	generateTemplateQuery,
+	IconMetadataFragFragmentDoc,
+	OpenGraphMetadataFragFragmentDoc,
+	QueryEngine,
+	RouteMetadataFragFragmentDoc,
+	SiteMetadataFragFragmentDoc,
+	TwitterMetadataFragFragmentDoc,
+} from '@snapwp/query';
+
+import {parseGeneralSettings as iconParser} from './icons-parser';
+import {parseGeneralSettings as siteParser} from './site-parser';
+
+import {parseNode as opengraphParser} from './opengraph-parser';
+import {parseNode as twitterParser} from './twitter-parser';
+import {parseNode as templateParser} from './template-parser';
+
+import type { Metadata } from 'next';
+import type { MetadataPlugin } from './types';
+
+/**
+ * Handles Seo related metadata.
+ */
+export class Seo {
+	static isInitialized: boolean = false;
+	static plugins: MetadataPlugin< any,any >[] = []
+    
+	/**
+	 * Initializer
+	 */
+	public static initialize(): void {
+		Seo.isInitialized = true;
+		Seo.registerDefaultPlugins();
+	}
+
+	/**
+	 * Loads a plugin to generate route level meta data.
+	 * @param {RootMetadataGeneratorPlugin< any >} plugin Plugin object
+	 */
+	public static registerPlugin(
+		plugin: MetadataPlugin< any,any >,
+	): void {
+		Seo.plugins.push(plugin);
+	}
+
+	/**
+	 * Loads all default plugins.
+	 */
+	private static registerDefaultPlugins(): void {
```

Comment:
Can we DRY this out with an array + a loop, instead of manually calling registerPlugin a bunch of times?